### PR TITLE
Fix maven javadoc generate

### PR DESCRIPTION
**Pull request by Shoothzj**
### Motivation
Currently, the maven javadoc generate is broken both on master and branch-4.14

### Changes
- fix the javadoc generate, the gradle script has already use `delombok`
- delete unneeded `package-info` file
- delete unused dependency `spotbugs-annotations`, it's not worth to open other pr(IMO)

### CODE REVIEW

**Comment by Shoothzj** [comment]
The javadoc is generated fine.
image

**Comment by Shoothzj** [comment]
rerun failure checks

**Comment by Shoothzj** [comment]
@dlg99 @eolivelli @nicoloboschi PTAL

**Comment by Shoothzj** [comment]
rerun failure checks

**Comment by Shoothzj** [comment]
@dlg99 @eolivelli @nicoloboschi PTAL

File: .github/workflows/pr-validation.yml [hunk 1]
```diff
@@ -56,6 +56,5 @@ jobs:
       - name: Check license files
         run: dev/check-all-licenses
 
-      # keeping on gradle, `mvn site javadoc:javadoc`, to figure out later
       - name: Generate Javadoc
-        run: ./gradlew generateApiJavadoc
+        run: mvn clean -B -nsu -am -pl :bookkeeper-common,:bookkeeper-server,:bookkeeper-stats-api,:bookkeeper-stats-providers,:codahale-metrics-provider,:prometheus-metrics-provider install javadoc:aggregate -DskipTests -Pdelombok
```
Comment (nicoloboschi) [inline .github/workflows/pr-validation.yml:60]: the Maven standard syntax for `-pl` does not require the colon, I'd prefer to remove them

File: stream/distributedlog/pom.xml [hunk 2]
```diff
@@ -47,8 +47,9 @@
         <artifactId>maven-javadoc-plugin</artifactId>
         <version>${maven-javadoc-plugin.version}</version>
         <configuration>
+          <sourcepath>${src.dir}</sourcepath>
+          <notimestamp>true</notimestamp>
           <!-- Avoid for missing javadoc comments to be marked as errors -->
```
Comment (nicoloboschi) [inline stream/distributedlog/pom.xml:52]: no longer useful comment

File: site3/website/scripts/javadoc-gen.sh [hunk 3]
```diff
@@ -34,7 +34,7 @@ function build_javadoc() {
   if [[ "$use_gradle" == "true" ]]; then
     ./gradlew generateApiJavadoc
   else
-    mvn clean install javadoc:aggregate -DskipTests
+    mvn clean -B -nsu -am -pl :bookkeeper-common,:bookkeeper-server,:bookkeeper-stats-api,:bookkeeper-stats-providers,:codahale-metrics-provider,:prometheus-metrics-provider install javadoc:aggregate -DskipTests -Pdelombok
```
Comment (nicoloboschi) [inline site3/website/scripts/javadoc-gen.sh:37]: the Maven standard syntax for -pl does not require the colon, I'd prefer to remove them

File: stream/distributedlog/pom.xml (hunk 2 above)
Comment (Shoothzj) [reply stream/distributedlog/pom.xml:52]: @nicoloboschi why? it's commented for `<doclint>none<doclint>`

**Comment by Shoothzj** [comment]
@nicoloboschi we can remove the colon of `bookkeeper-common` and `bookkeeper-server`. We need colon for non-top-level modules. I have pushed a fix.

File: pom.xml [hunk 4]
```diff
@@ -110,6 +110,7 @@
     <javac.target>1.8</javac.target>
     <redirectTestOutputToFile>true</redirectTestOutputToFile>
     <testRetryCount>2</testRetryCount>
+    <src.dir>src/main/java</src.dir>
```
Comment (eolivelli) [inline pom.xml:113]: if this is only for "delombok" then what about giving a more specific name ? like output.delombok.dir ?

**Review by eolivelli — APPROVED** [review]
LGTM

I left one suggestion

File: pom.xml (hunk 4 above)
Comment (Shoothzj) [reply pom.xml:113]: it's also used by javadoc maven plugin.

**Review by nicoloboschi — APPROVED** [review]

**Comment by hangc0276** [comment]
@Shoothzj This Pr has too many conflicts with branch-4.14, would you please push a separate PR to branch-4.14?

**Comment by Shoothzj** [comment]
@hangc0276 I am on vacation now. @xiang092689 He volunteered to help with this.

**Pull request merged**